src/backend/distributed/utils/background_jobs.c

```diff
@@ -706,8 +706,12 @@ TaskConcurrentCancelCheck(TaskExecutionContext *taskExecutionContext)
 	BackgroundExecutorHashEntry *handleEntry = taskExecutionContext->handleEntry;
 	BackgroundTask *task = GetBackgroundTaskByTaskId(handleEntry->taskid);
 	taskExecutionContext->task = task;
+	if (!task)
+	{
+		ereport(ERROR, (errmsg("task is invalid")));
+	}
 
-	if (!task || task->status == BACKGROUND_TASK_STATUS_CANCELLING)
+	if (task->status == BACKGROUND_TASK_STATUS_CANCELLING)
```

Comment:
that's a bit of a trouble, in the "else" part too it depends on task being set.

Reply:
if exit the function earlier, it will not be troble